kubedb.com/v1/redis.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright AppsCode Inc. and Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"fmt"
+
+	"kmodules.xyz/resource-metrics/api"
+
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func init() {
+	api.Register(schema.GroupVersionKind{
+		Group:   "kubedb.com",
+		Version: "v1",
+		Kind:    "Redis",
+	}, Redis{}.ResourceCalculator())
+}
+
+type Redis struct{}
+
+func (r Redis) ResourceCalculator() api.ResourceCalculator {
+	return &api.ResourceCalculatorFuncs{
+		AppRoles:               []api.PodRole{api.PodRoleDefault},
+		RuntimeRoles:           []api.PodRole{api.PodRoleDefault, api.PodRoleExporter},
+		RoleReplicasFn:         r.roleReplicasFn,
+		ModeFn:                 r.modeFn,
+		UsesTLSFn:              r.usesTLSFn,
+		RoleResourceLimitsFn:   r.roleResourceFn(api.ResourceLimits),
+		RoleResourceRequestsFn: r.roleResourceFn(api.ResourceRequests),
+	}
+}
+
+func (r Redis) roleReplicasFn(obj map[string]interface{}) (api.ReplicaList, error) {
+	mode, found, err := unstructured.NestedString(obj, "spec", "mode")
+	if err != nil {
+		return nil, err
+	}
+	if found && mode == DBModeCluster {
+		shards, _, err := unstructured.NestedInt64(obj, "spec", "cluster", "master")
```

Comment:
This should be (obj, "spec", "cluster", "shards"). Can you check @Neaj-Morshad-101

Reply:
yes, Updated.